pkg/postage/listener/listener.go

```diff
@@ -0,0 +1,264 @@
+package listener
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/postage"
+)
+
+const (
+	blockPage = 500 // how many blocks to sync every time
+	tailSize  = 100 // how many blocks to tail from the tip of the chain
+)
+
+var (
+	chainUpdateInterval = 30 * time.Second
+)
+
+type BlockHeightContractFilterer interface {
+	bind.ContractFilterer
+	BlockNumber(context.Context) (uint64, error)
+}
+
+type listener struct {
+	logger          logging.Logger
+	ev              BlockHeightContractFilterer
+	postageStampABI abi.ABI
+	priceOracleABI  abi.ABI
+
+	batchCreatedTopic       common.Hash
+	batchTopupTopic         common.Hash
+	batchDepthIncreaseTopic common.Hash
+	priceUpdateTopic        common.Hash
+
+	postageStampAddress common.Address
+	priceOracleAddress  common.Address
+
+	quit chan struct{}
+	wg   sync.WaitGroup
+}
+
+func New(
+	logger logging.Logger,
+	ev BlockHeightContractFilterer,
+	postageStampAddress,
+	priceOracleAddress common.Address,
+) postage.Listener {
+	postageStampABI := parseABI(PostageStampABI)
+	priceOracleABI := parseABI(PriceOracleABI)
+	return &listener{
+		logger:                  logger,
+		ev:                      ev,
+		postageStampABI:         postageStampABI,
```

Comment:
the ABIs and topics don't need to be fields. We could just go with the same approach like in `listener_test.go`, where we have these as package level `var`s. After this is done, we can just expose the ones here through `export_test` and remove the ones in `listener_test.go`

Reply:
done